### DISCUSSION TITLE
What default desktop switching mechanism do you want?

### DISCUSSION THREAD

**Discussion by love-code-yeyixiao** [sandboxie-plus/Sandboxie#4019] [Polls]
We plan to release sandbox desktop features in two or three Latest releases. What default sandbox switching mechanism do you want?

**Comment by bastik-1001** [comment]
Can you please explain, what this feature is going to do? I have not voted, yet. The second and the fifth option seem to be messy, but maybe I fail to understand what the feature is going to do.

**Comment by offhub** [comment]
https://www.wilderssecurity.com/threads/sandboxed-desktop-new-feature-work-in-progress.454168/

**Comment by xsmolasses** [comment]
Is this implementation aligned and accessible with the virtual Desktops in Windows 10/11 Task View? If not, keep the new Sbie API exposed for use by third-party programs; for service-like speed in switching.

I’d like to see the Windows Shell / Taskbar reinvented someday, using a lite web-based engine, probably Qt something or other; and to bring back the [Windows 95/98(/XP 32-bit)] "Active Desktop", too. Endless customisation!

In the interim, Start.exe could tack on a new /parameter (itself making use of said API) for ease of integration by users' own needs to switch Desktops using preferred launcher (e.g., Elgato Stream _Dreck_) by way of simple open/run target.

**Reply by love-code-yeyixiao** [reply]
Sandboxed desktop is a real desktop instead of a "virtual desktop",it has a level the same as "WinLogon Desktop" or "Windows Secure Desktop",so it's impossible to compital with native TaskBar like virtual desktops.

The purpose of the sandboxed desktop feature is isoation,so we won't allow applications to switch desktop without users' permit.

So far,we are still finding third-party open source shell project, so maybe we will not make by ourselves if we find a fit one.

Now we have already make "KmdUtil.exe" be able to switch desktops by commandline,you can see deatils by seeing branch "Experiment".

**Reply by e-t-l** [reply]
Does the isolation here work with the same technology as Secure Desktop, like is used for UAC?

**Reply by love-code-yeyixiao** [reply]
> Does the isolation here work with the same technology as Secure Desktop, like is used for UAC?

Yes.

**Comment by bastik-1001** [comment]
It's good to have a default (whatever that might be in the end), but please, make it possible to disable automatic switching. That way, users can disable it and switch manually, if they wish so.
